src/routes/safe/components/Transactions/TxList/hooks/useTransactionType.ts

```diff
@@ -37,10 +41,37 @@ export const useTransactionType = (tx: Transaction): TxTypeProps => {
         break
       }
       case 'SettingsChange': {
-        // deleteGuard doesn't exist in Solidity
-        // It is decoded as 'setGuard' with a settingsInfo.type of 'DELETE_GUARD'
-        const isDeleteGuard = tx.txInfo.settingsInfo?.type === 'DELETE_GUARD'
-        setType({ icon: SettingsTxIcon, text: isDeleteGuard ? 'deleteGuard' : tx.txInfo.dataDecoded.method })
+        const { settingsInfo, dataDecoded } = tx.txInfo
+        const icon = SettingsTxIcon
+
+        switch (settingsInfo?.type) {
+          case 'DELETE_GUARD': {
+            // deleteGuard doesn't exist in Solidity
+            // It is decoded as 'setGuard' with a settingsInfo.type of 'DELETE_GUARD'
+            setType({ icon, text: 'deleteGuard' })
+            break
+          }
+          case 'CHANGE_THRESHOLD': {
+            setType({ icon, text: `${dataDecoded.method} (${settingsInfo.threshold}/n)` })
+            break
+          }
+          case 'ADD_OWNER':
+          case 'REMOVE_OWNER': {
+            const newThreshold = settingsInfo.threshold
+            const method = settingsInfo.type === 'ADD_OWNER' ? 'addOwner' : 'removeOwner'
+
+            // We only have the current threshold as reference, therefore threshold/n
```

Comment:
We cannot determine the 'historical' threshold as it is not returned alongside the transaction.